README.md

```diff
@@ -1,18 +1,334 @@
+
+To retrieve a declared variable, refer to its name:
+```js
+pi;
+//=> 3.14159
+```
+
+### Reassigning Variable Value
+
+Changing the value of a variable in JavaScript works just in the same way
+as it does in Ruby:
+
+```js
+var pi = 3.14159;
+pi;
+//=> 3.14159
+pi = 3.14;
+pi;
+//=> 3.14;
+```
+
+### Variable Values
+
+Upon declaration, all variables are automatically assigned the value of
+`undefined`. It's only after we assign a new value that the variable will
+contain something other than undefined. We can use the `typeof` operator to
+check the data type of the value currently stored in a variable:
+
+```js
+var language;
+//=> undefined
+
+typeof language;
+//=> "undefined"
+
+language = "JavaScript";
+//=> "JavaScript"
+
+typeof language;
+//=> "string"
+```
+
+***Top Tip***: When writing JavaScript code, it's good practice to ***never***
+set a variable equal to `undefined`. Variables will be `undefined` until we
+explicitly assign a value, so encountering an `undefined` variable is a strong
+signal that the variable was declared but not assigned prior to the reference.
+That's valuable information that we can use while debugging, and it comes at no
+additional cost to us.
+
+Once a variable has been created with `var`, we can reassign it to our heart's
+content:
+
+```js
+var pi = 3.14159;
+//=> undefined
+
+typeof pi;
+//=> "number"
+
+pi = "the ratio between a circle's circumference and diameter";
+//=> "the ratio between a circle's circumference and diameter"
+
+typeof pi;
+//=> "string"
+```
+
+The data that's stored in our variable might change over time, but at any moment
+we can retrieve its current contents:
+
+```js
+var language = "Mocha";
+//=> undefined
+
+language = "LiveScript";
+//=> "LiveScript";
+
+language = "JavaScript";
+//=> "JavaScript";
+
+language;
+//=> "JavaScript";
+```
+
+## Identify When to Use `const`, `let`, and `var` for Declaring Variables
+
+Because of its ubiquity in legacy code and StackOverflow posts, it's important
+to get to know `var`. However, as we alluded to earlier, there is almost no
+reason to use `var` with the features JavaScript has post-2015. `var` comes with
+a ton of baggage in the form of scope issues (which we will discuss in the
+lesson on scope in JavaScript) and allows developers to play a little too
+fast and loose with variable declarations.
+
+For example, with `var`, no error is thrown if you declare a variable twice:
+
+```js
+var language = "Ruby";
+//=> undefined
+
+var language = "JavaScript";
+//=> undefined
+
+language;
+//=> "JavaScript"
+```
+
+This is bad! There's no reason to declare a variable twice, and it's usually a
+mistake by a developer unaware that the variable had already been declared.
+
+### `let`
+
+ES2015 introduced two new ways to create variables: `let` and `const`. Both
+solve all of `var`'s scope issues, which, again, we'll cover in the lesson on
+scope in JavaScript. Both also throw an error if you try to declare the same
```

Comment:
Also remove this reference. Maybe we can replace it with something vaguer, like "which is whole other topic" or "which we cover in a different section._ We just don't want to promising something that might not be part of someone's curriculum package.

Reply:
Agreed. Some of the language came from pulling from existing lessons.